### DISCUSSION TITLE
Creating an EditProfile and ChangePassword page in filamentphp and using this page in all panels (admin, user, ...)

### DISCUSSION THREAD

**Discussion by HamidGj** [filamentphp/filament#13058] [Help] [answered]
### Package

Panel builder

### Package Version

v3

### How can we help you?

I want to use an editProfile and ... page for all panels, but when creating the page, the type of panel must be specified.

**Which panel would you like to create this in? [user]
 admin, user, agent**

If the admin panel is selected when creating the page, when adding the following code to the user panel:
```
->userMenuItems([
 'profile' => MenuItem::make()
 ->url(fn(): string => EditProfile::getUrl())
 ])
```
I get the following error.

> Symfony \ Component \ Routing \ Exception \ RouteNotFoundException
> PHP 8.3.7
> 11.1.1
> Route [filament.user.pages.edit-profile] not defined.
> 

If I define the path of the view in the routes like below
```
Route::get('user/edit-profile', function(){
 return view('filament.pages.edit-profile');
})->name('filament.user.pages.edit-profile');
```
 When I click on the edite profile item the website, I get the following error

> Error
> PHP 8.3.7
> 11.1.1
> Using $this when not in object context
> 

```
<x-filament-panels::page>
 <x-filament-panels::form wire:submit="updateProfile" >

 {{ $this->editProfileForm }}

 <x-filament-panels::form.actions
 :actions="$this->getUpdateProfileFormActions()"
 />
 </x-filament-panels::form>
</x-filament-panels::page>
```
Is there a solution that uses one page or should I create a separate page for each panel?

**Comment by leandrocfe** [comment, marked as answer]
If you are using V3, you can use the profile page in both panels. Just add `profile()` method in the providers

```php
$panel
->profile()
```

**Reply by HamidGj** [reply]
Thank.
How do I change my profile picture?
and how to added password validation rule in the validator?

**Reply by leandrocfe** [reply]
You can create a new profile class, extend the default profile class and edit/override what you need

```php
$panel
->profile(MyCustomProfile::class)
```

**Reply by bilogic** [reply]
I have `->login(null)->profile()` and getting `Route [filament.admin.auth.profile]` not defined.
